client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -184,7 +184,15 @@
      */
     @Beta
     CompletableFuture<Map<Stream, StreamCut>> generateStreamCuts(ScheduledExecutorService backgroundExecutor);
-    
+
+    /**
+     * Returns current distribution of number of segments assigned to each reader in the reader group. 
+     * For unassigned segments, it adds an entry to the map where readerId is empty string.
+     *
+     * @return A map of readerId to segment count.   
+     */
+    Map<String, Integer> getReaderSegmentDistribution();
```

Comment:
We can move this API to `io.pravega.client.stream.ReaderGroupMetrics`. Since this API is similar to the `io.pravega.client.stream.ReaderGroupMetrics#unreadBytes` in the sense that it provides insight into the current metric/state of the ReaderGroup.

Reply:
I had thought about it. However, i think metrics is a single numerical value. 
I dont know if the distribution really fits into that. Hence i kept it outside of metrics.